### DISCUSSION TITLE
How do I build boost using vcpkg targeting Windows 10?

### DISCUSSION THREAD

**Discussion by FrankHeimes** [microsoft/vcpkg#23612] [Q&A] [answered]
When I attempt to link to `boost_log-vc140-mt.lib` after building it on Windows 10, I get unresolved external symbol errors.

The cause is that `boost_log-vc140-mt.lib` contains functions in namespace `v2s_mt_nt6` but my client application expects functions in namespace `v2s_mt_nt62`.

This indicates that neither `_WIN32_WINNT` nor `WINVER` were defined while boost was built by vcpkg, causing `vcpkg\installed\x64-windows-static\include\boost\log\detail\config.hpp` to revert to Windows 7 as the target version. This causes `BOOST_LOG_VERSION_NAMESPACE` to be defined as `v2s_mt_nt6`.

When building my client application, `_WIN32_WINNT` is defined as `0x0A00` (Windows 10), causing `BOOST_LOG_VERSION_NAMESPACE` to be defined as `v2s_mt_nt62`. Hence that definition doesn't match the one used to build the static library `boost_log-vc140-mt.lib`.

What is the correct way to configure vcpkg so that it builds all projects targeting Windows 10, i. e. how can I enforce that `_WIN32_WINNT == 0x0A00` each time vcpkg calls cmake?

I didn't find any clues in the port files for boost.

Boost 1.78.0, Visual Studio 2022, 17.1.1, Toolset v143, Windows 10 1809

**Comment by autoantwort** [comment, marked as answer]
It should be possible to set `VCPKG_CXX_FLAGS` in a custom triplet. Then you build boost using this custom triplet

**Reply by FrankHeimes** [reply]
Thanks for answering! Yes this works. To be more precise, I added this to the triplet:
```
set(VCPKG_C_FLAGS -D_WIN32_WINNT=0x0A00)
set(VCPKG_CXX_FLAGS -D_WIN32_WINNT=0x0A00)
```
